src/NLog/Targets/Target.cs

```diff
@@ -722,6 +724,39 @@ protected string RenderLogEvent(Layout layout, LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Renders the event info in layout.
+        /// </summary>
+        /// <param name="layout">The layout.</param>
+        /// <param name="logEvent">The event info.</param>
+        /// <param name="defaultValue">The default value when <paramref name="layout"/> and/or <paramref name="logEvent"/> is <c>null</c></param>
+        /// <returns>Value representing log event.</returns>
+        protected T RenderLogEvent<T>([CanBeNull] Layout<T> layout, [CanBeNull] LogEventInfo logEvent, T defaultValue = default(T))
+        {
+            if (layout == null || logEvent == null)
+                return defaultValue;
+
+            if (OptimizeBufferReuse)
+            {
+                if (layout.IsFixed)
+                {
+                    return layout.RenderToValueOrDefault(logEvent, defaultValue);
```

Comment:
Should just call `layout.RenderToValue(logEvent, default);`

Reply:
good one, but then I also need `     IRenderable<T> renderable = l;`. This makes the code more complex. Is that really worth it?

Reply:
Why is that cast necessary to call a method. You have the full type `Layout<T>`

Reply:
Tis is not possible, as RenderToValue is on `IRenderable<T>` and `Layout<T> ` implements it explictly: ` T IRenderable<T>.RenderToValue(LogEventInfo logEvent, T defaultValue)`.

This is by design, so that others prefer `RenderToValueOrDefault` over `RenderToValue`